egs/hub4_spanish/s5/local/prepare_unicode_lexicon.py

```diff
@@ -0,0 +1,198 @@
+        base_graph = g.split("_")[0]
+        if(base_graph not in graphemes_dict.keys()):
+            graphemes_dict[base_graph] = []
+
+        graphemes_dict[base_graph].append(g)
+    
+    return unicode_graphemes, graphemes_dict
+
+
+def write_nonsilence_phones(graphemes_dict, nonsilence_phones,
+                            extraspeech=None):
+    with codecs.open(nonsilence_phones, "w", "utf-8") as fp:
+        try:
+            with codecs.open(extraspeech, "r", "utf-8") as f:
+                for line in f:
+                    line_vals = line.strip().split()
+                    fp.write("%s\n" % line_vals[1])
+        except (IOError, TypeError):
+            pass 
+
+        # Write each base grapheme with all tags on the same line
+        for base_grapheme in sorted(graphemes_dict.keys()):
+            line = ""
+            for grapheme in graphemes_dict[base_grapheme]:
+                line += grapheme + " "
+            fp.write("%s\n" % line.strip())
+
+
+def write_extra_questions(unicode_graphemes, graphemes_dict, tags,
+                          extra_questions, nonspeech=None, extraspeech=None):
+    with codecs.open(extra_questions, "w", "utf-8") as fp:
+        # Write all unique "phones" but graphemes in this case, plus <hes> to a
+        # single line.
+        
+        # Write the extraspeech
+        try:
+            with codecs.open(extraspeech, "r", "utf-8") as f:
+                for line in f:
+                    line_vals = line.strip().split()
+                    fp.write("%s " % line_vals[1])
+        except (IOError, TypeError):
+            pass
+        
+        for g in unicode_graphemes:
+            fp.write("%s " % g)
+        fp.write("\n")
+        
+        # Write the nonspeech
+        try:
+            with codecs.open(nonspeech, "r", "utf-8") as f:
+                for line in f:
+                    line_vals = line.strip().split()
+                    fp.write("%s " % line_vals[1])
+                fp.write("\n")
+        except (IOError, TypeError):
+            pass
+
+        # Write all possible phone_tag combinations that occur in the lexicon
+        for tag in tags:
+            for g in graphemes_dict.keys():
+                tagged_grapheme = "_".join([g, tag])
+                if(tagged_grapheme in graphemes_dict[g]):
+                    fp.write("%s " % tagged_grapheme)
+            fp.write("\n")
+
+
+def main():
+    #  --------------- Extract unicode_graphemes from the table --------------
+    if(len(sys.argv[1:]) == 0):
+        print("Usage: local/prepare_unicode_lexicon.txt <lex_table>"
+              " <lex_dir>" )
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("table", help="Table containing all information about"
+                        " how to map unicode graphemes to unicode descriptors")
+    parser.add_argument("lex_dir", help="Directory to which all files"
+                        " should be written")
```

Comment:
This gives the impression that lex_dir is omly an input, but aren't you getting some information from there too?